sphinx_gallery/docs_resolv.py

```diff
@@ -232,8 +233,15 @@ def _get_link(self, cobj):
                 fname_idx = value[cobj['name']][0]
 
         if fname_idx is not None:
-            fname = self._searchindex['filenames'][fname_idx] + '.html'
-
+            fname = self._searchindex['filenames'][fname_idx]
+            try:
+                import sphinx
+            except Exception:
```

Comment:
It looked like nowhere in the package did `sphinx` actually get imported, so I didn't want to add a top-level import. But since this is `sphinx-gallery` I guess it's okay, eh? :)